### PR TITLE
doc: section reorganization

### PR DESCRIPTION
**Pull request by annwoj**
removed the content duplicated with the Amazon docs

### CODE REVIEW

File: doc/setting_up_sidewalk_environment/setting_up_sidewalk_gateway.rst [hunk 1]
```diff
@@ -1,81 +1,25 @@
...
-   Your accounts are now linked.
-
-   .. figure:: /images/sidewalk_ring_image2.png
+For the exact instructions, refer to the `Setting up a Sidewalk gateway`_ documentation.
 
 .. note::
    If you have multiple Echo devices registered under a single Amazon account, only one of those devices will be able to operate as an FSK gateway even if other devices have the same capability.
    If you need to operate multiple FSK gateways, we recommend setting them up under separate Amazon accounts.
 
+.. TBC: "Linking to your ring account" was included in our documentation but does not appear after Amazon's "setting up a Sidewalk gateway". Should the instruction be kept on our side, or should we refer to this instead - https://docs.sidewalk.amazon/mobile-sdk/developer-guide/sidewalk-mobile-sdk-overview/#what-is-amazon-sidewalk-mobile-sdk? Or maybe drop it completely?
```
Comment (annwoj) [inline doc/setting_up_sidewalk_environment/setting_up_sidewalk_gateway.rst:19]: @wojciechpluszynski - question to this section (line 44 in the previous file)

File: doc/setting_up_sidewalk_environment/setting_up_sidewalk_product.rst [hunk 2]
```diff
@@ -5,6 +5,8 @@ Setting up your Sidewalk product
 
 To complete the set up of your Sidewalk product, you have to generate provisioning and add MQTT to destination.
 
+.. TBC: Which parts of the article should be kept. Should it be left as is? Additionally, Amazon repository has Nordic-specific instructions (https://github.com/aws-samples/aws-iot-core-for-amazon-sidewalk-sample-app/blob/main/EdgeDeviceBinaries/nordic/doc/_How_to_program.md ), are these valid?
```
Comment (annwoj) [inline doc/setting_up_sidewalk_environment/setting_up_sidewalk_product.rst:8]: @wojciechpluszynski - question here as well

Comment (annwoj) [reply doc/setting_up_sidewalk_environment/setting_up_sidewalk_product.rst:8]: confirmed - comment to be removed for now

File: doc/setting_up_sidewalk_environment/setting_up_sidewalk_gateway.rst (hunk 1 above)
Comment (wojciechpluszynski) [reply doc/setting_up_sidewalk_environment/setting_up_sidewalk_gateway.rst:19]: This step is not required when setting up new Echo device and Sidewalk end device. Let's remove from Nordic documentation.

**Pull request merged**